src/librustc_mir/borrow_check.rs

```diff
@@ -934,43 +932,53 @@ impl<'c, 'b, 'a: 'b+'c, 'gcx, 'tcx: 'a> MirBorrowckCtxt<'c, 'b, 'a, 'gcx, 'tcx>
     fn report_conflicting_borrow(&mut self,
                                  _context: Context,
                                  (lvalue, span): (&Lvalue, Span),
-                                 loan1: (&Lvalue, BorrowKind),
-                                 loan2: (&Lvalue, BorrowKind)) {
-        let (loan1_lvalue, loan1_kind) = loan1;
-        let (loan2_lvalue, loan2_kind) = loan2;
+                                 gen_borrow_kind: BorrowKind,
+                                 issued_borrow: &BorrowData) {
         // FIXME: obviously falsifiable. Generalize for non-eq lvalues later.
-        assert_eq!(loan1_lvalue, loan2_lvalue);
+        assert_eq!(lvalue, &issued_borrow.lvalue);
+
+        let issued_span = self.retrieve_borrow_span(issued_borrow);
+
+        let issued_end_loan_span = if let Lvalue::Local(local) = issued_borrow.lvalue {
```

Comment:
That's not how you do it. You should look up the span of the `EndRegion`.

Reply:
I've spent some time trying to do this, but `StatementKind::EndRegion` contains `region::Scope`, and I cannot get span from it. ```
impl Scope {
 pub fn span(&self, tcx: TyCtxt, scope_tree: &ScopeTree)
```
Where should I get ScopeTree? I found MIR build already do this and resided to use it result.

Reply:
@mikhail-m1 I think the idea is that the `EndRegion` *itself* is a `Statement`, and every `Statement` carries a `SourceInfo` which in turn carries a `Span`.